storey/sources.py

```diff
@@ -875,3 +875,111 @@ def _init(self):
             else:
                 df = pandas.read_parquet(path, columns=self._columns, storage_options=self._storage_options)
             self._dfs.append(df)
+
+    class MongoDBSource(_IterableSource, WithUUID):
+        """Use mongodb collection as input source for a flow.
+
+        :parameter key_field: column to be used as key for events. can be list of columns
+        :parameter time_field: column to be used as time for events.
+        :parameter id_field: column to be used as ID for events.
+        :parameter db_name: the name of the database as mention on mongodb
+        :parameter connection_string:
+        :parameter collection_name: the name of the collection as mention on mongodb
+
+        for additional params, see documentation of  :class:`~storey.flow.Flow`
+        """
+
+        def __init__(
+                self,
+                db_name: str = None,
+                connection_string: str = None,
+                collection_name: str = None,
+                query: dict = None,
+                key_field: Optional[Union[str, List[str]]] = None,
+                start_filter: Optional[datetime] = None,
+                end_filter: Optional[datetime] = None,
+                time_field: Optional[str] = None,
+                id_field: Optional[str] = None,
+                **kwargs,
+        ):
+
+            if end_filter or start_filter:
+                start_filter = datetime.min if start_filter is None else start_filter
+                end_filter = datetime.max if end_filter is None else end_filter
```

Comment:
Can simplify as
```
                start_filter = start_filter or datetime.min
                end_filter = end_filter or datetime.max
```
But do you even have to add these to the dictionary if they're not set?